src/addons/matchrecursive.js

```diff
@@ -179,10 +180,19 @@ export default (XRegExp) => {
                         break;
                     }
                 }
-            } else {
+            } else if (unbalancedDelimiters === 'error') {
                 const delimSide = rightMatch ? 'right' : 'left';
                 const errorPos = rightMatch ? delimStart : outerStart;
                 throw new Error(`Unbalanced ${delimSide} delimiter found in string at position ${errorPos}`);
+            } else if (unbalancedDelimiters === 'text') {
+                if (rightMatch) {
+                    rightMatch = null;
+                } else {
+                    delimEnd = XRegExp.exec(str, left, outerStart)[0].length + outerStart;
```

Comment:
Note the edge case where `left` matches an empty string. Is this already covered by the empty string delimiter handling below (`if (delimStart === delimEnd) {...}`)? If not, it needs to be covered here (`delimEnd = outerStart + (<...>.length || 1)`).

---
I think this is the correct handling -- specifically, advancing to the end of the outer left delimiter instead of advancing by one character (with a more simple `delimEnd = outerStart + 1`). But it's a choice we're making.

Consider how this can change behavior when the left delimiter is able to match at least two characters. For a simplified example, I'll use the subject string `'<<<<>>'`, left delimiter `'<<'`, and right delimiter `'>>'`, but keep in mind that the regex delimiters could be much more dynamic in what they match.

By advancing here to the end of the left delimiter, you get a _between_ value of `'<<'`, _left_ `'<<'`, _match_ `''`, and _right_ `'>>'`. However, if you were to advance by one character instead, you would get a _between_ value of `'<'`, _left_ `'<<'`, _match_ `'<'`, and _right_ `'>>'`.

I suspect your current handling will be more intuitive in all but extremely edge cases, so I think it's probably the right way to go by default. But it opens up the option of a future handling mode (I'll call it `'skip-lazy'` for now) that advances by one character when it finds an unbalanced delimiter.

Reply:
ooh that's interesting. I'll note that the implementation is heavily biased towards my intended use case 😄.

Here's another edge case that this made me think of: `<<<>>`

Intuitively, one might expect a match on the last four characters. however, I believe the current implementation would skip over the whole string entirely bc outer left would increment to the substring `<>>`.  This would also be handled by your `skip-lazy` idea, but just wanted to make note of it

Reply:
Yeah, that's a great additional example. I wonder if it would be better to include both `'skip'` and `'skip-lazy'` modes right off the bat to give people the choice.

Reply:
Turns out my example `<<<>>` was treated the same in both cases, with the middle `<` as the match. Your example showcased the difference so I used that one for the test